### ISSUE TITLE
Incorrect processing/clipping of lines

### ISSUE THREAD

**Issue by patrikkarlsson817** [mapbox/mapnik-vector-tile#23]
It seems like the vector::processor sometimes fails to get some lines (from a shapefile) on the buffered part of the tile, especially when the line seems to be north/south of the tile. 

I have rendered all geometries in some tiles and the resulting image shows that some smaller roads and (which are more problematic for my case) label-lines are removed in the buffered part (compared to the surrounding tiles actual data/image). This results in cut off labels (the line is the center of the text) for some tiles.

I have tried to take a larger buffer but it doesn´t fix it. Is this a known/unknown problem or have I misunderstood something?

**Comment by knutole** [comment]
Hi. Wondering if I have the same problem - or if it's different. Rendering tiles from vector_tiles (made with geojson extracted from OSM PostGIS). For example roads are "clipped" rather weirdly at the tile-edges. So are buildings (when drop-shadow is applied). 
![screen shot 2015-01-28 at 14 09 50](https://cloud.githubusercontent.com/assets/2197944/5938049/89336c98-a6f7-11e4-829b-db31979f2897.png)
 
Is this due to `buffer_size` settings? I've tried all kinds of `buffer_size` settings (on `vtile.render()`), but the only settings that have visible effects are negative values (like `-64`). 

Long story short, how do I go about fixing this? Is it due to lacking `buffer_size` in the `vector_tiles` themselves, yielding no difference in rasters? Any pointers greatly appreciated!

**Comment by durandt** [comment]
Hi, 

I remember struggling a bit with that a few months ago. 
One hint that may help you: 

You have to think about the whole flow when dealing with this issue: 
1. Do your vectortiles contain the data outside of the BBox? (ie: Do you have a buffe when generating the tiles? Buffer size may depend on the type of data you need to render)
2. During rendering, mapnik queries the data from the tile, make sure that the buffer is also applied there. 
Components are nested in each other so check how the data is extracted from the tile at every level before it is rendered. 
You have to distinguish the area that you query the data from, and the area that you want to create an image for. 

Hope that helps a bit.

**Comment by knutole** [comment]
Hi @durandt , thanks!  Indeed, the `vtile` does not have a `buffer` around it - and that is probably an issue. I'm going to try to add a buffer for the `vtile` also, so that the raster render has something to work with in the buffer zone. Thanks for your input!

**Comment by knutole** [comment]
Indeed, putting a buffer on the `vector_tile` helps! Thanks!

**Comment by flippmoke** [comment]
I am closing this ticket as I believe it is resolved. Additionally, the code base has changed quite a bit since this ticket was opened!
